streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -134,9 +134,7 @@ private static FileLock lockStateDirectory(File stateDir, int retry) throws IOEx
             retry--;
             lock = lockStateDirectory(channel);
         }
-        if (lock == null) {
-            channel.close();
-        }
+
         return lock;
```

Comment:
It seems like this method returns `null` if do not manage to `lock` after `retry` attempts. Probably a good idea to document that in the public `lockStateDirectory`. Reading the current documentation, I would have expected an exception to be thrown.